src/audio/module_adapter/module/cadence.c

```diff
@@ -862,8 +862,7 @@ cadence_codec_set_configuration(struct processing_module *mod, uint32_t config_i
 		return ret;
 
 	/* return if more fragments are expected or if the module is not prepared */
-	if ((pos != MODULE_CFG_FRAGMENT_LAST && pos != MODULE_CFG_FRAGMENT_SINGLE) ||
-	    md->state < MODULE_IDLE)
+	if (pos != MODULE_CFG_FRAGMENT_LAST && pos != MODULE_CFG_FRAGMENT_SINGLE)
```

Comment:
@ranj063 @lgirdwood comment on here? I remember seeing discussions where the module common code should be checking the module state machine is followed, and removing these checks seems to go counter to that.